### PR TITLE
Add hint for new dependencies required for Flyway

### PR DESCRIPTION
**Pull request by jmewes**
After following the documentation in https://docs.spring.io/spring-boot/how-to/data-initialization.html#howto.data-initialization.migration-tool.flyway for the setup of Flyway, I got the following error after attempting the app startup:

> org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Unsupported Database: PostgreSQL 16.3

It turns out that with Spring Boot 3.3, Flyway was upgraded to version 10 which requires an additional dependency for Posgres and some other databases:

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.3-Release-Notes#flyway-10

This PR is intended to provide a hint about this additional dependency in the documentation linked above.

Also see https://github.com/openrewrite/rewrite-spring/issues/532

### CODE REVIEW

File: spring-boot-project/spring-boot-docs/src/docs/antora/modules/how-to/pages/data-initialization.adoc [hunk 1]
```diff
@@ -98,6 +98,7 @@ Spring Boot supports two higher-level migration tools: https://flywaydb.org/[Fly
 === Execute Flyway Database Migrations on Startup
 
 To automatically run Flyway database migrations on startup, add the `org.flywaydb:flyway-core` to your classpath.
+For some databases, a https://search.maven.org/search?q=g:org.flywaydb%20a:flyway-database-*[database-specific module] needs to be added additionally, e.g. `org.flywaydb:flyway-database-postgresql` for PostgreSQL.
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-docs/src/docs/antora/modules/how-to/pages/data-initialization.adoc:101]: This wildcard search doesn't work for me and reports 0 results. I think we may have to link to https://documentation.red-gate.com/flyway/flyway-cli-and-api/supported-databases instead. There are also some DB-specific modules that are not named `flyway-database-*` such as the [module for MySQL](https://documentation.red-gate.com/flyway/flyway-cli-and-api/supported-databases/mysql).

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks for the PR, @jmewes. I've left a comment for your consideration when you have a minute.

**Comment by jmewes** [comment]
In this blog post a comprehensive list of the new dependencies can be found:

https://github.com/flyway/flyway/issues/3780

File: spring-boot-project/spring-boot-docs/src/docs/antora/modules/how-to/pages/data-initialization.adoc [hunk 2]
```diff
@@ -98,6 +98,7 @@ Spring Boot supports two higher-level migration tools: https://flywaydb.org/[Fly
 === Execute Flyway Database Migrations on Startup
 
 To automatically run Flyway database migrations on startup, add the `org.flywaydb:flyway-core` to your classpath.
+All databases that are not in-memory or file based need an additional dependency, e.g. `org.flywaydb:flyway-database-postgresql` is requiredd for PostgreSQL and `org.flywaydb:flyway-mysql` is required for MySQL (see https://documentation.red-gate.com/flyway/flyway-cli-and-api/supported-databases[Supported Databases in the Flyway Documentation] for details).
```
Comment (jmewes) [inline spring-boot-project/spring-boot-docs/src/docs/antora/modules/how-to/pages/data-initialization.adoc:101]: > As a result, databases, with the exception of in-memory or file databases such as H2 or SQLite, have been moved into their own packages which need to be added to your project when upgrading to Flyway V10.0.0. This includes whether you are using Flyway as a dependency in your project or using the Gradle or Maven plugins. -- https://github.com/flyway/flyway/issues/3780

**Comment by jmewes** [comment]
@wilkinsona,

Thanks for the feedback. I have added another commit with an attempt to clarify the required dependencies.

**Pull request closed without merging**